### PR TITLE
feat/one-page-ui

### PR DESCRIPTION
**Pull request by sanjay270899**
# One Page UI

[Notion Task Link](https://www.notion.so/One-Page-UI-for-all-Algo-content-00c7745bff884c06a7a43f4bdf1d875c)

#### Branches for testing

Frontend: `sanjay/one-page-ui`

Backend: `one-page-ui`

Bot: `NA`

#### Linked pull requests

[Backend](https://github.com/devs-nest/backend/pull/77)

# Feature Screenshot

image
image

### CODE REVIEW

File: src/components/Videos/Question.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      status_ = 'doubt';
+    } else if (status === 'doubt') {
+      // 2
+      status = 0; // done
+      status_ = 'done';
+    } else if (status === 'done') {
+      // 0
+      status = 1; // notdone
+      status_ = 'notdone';
+    }
```
Comment (patheticGeek) [inline src/components/Videos/Question.js:41]: Move to the `submitQuestion` service, take status in both number and string and then inside it convert to appropriate format

File: src/components/Videos/Video.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+              {video.title}
+            </span>
+          </span>
+          <span className="bg-white rounded p-1 shadow-sm">{video.tag}</span>
+        </div>
+        <div
+          className={`mt-3 rounded overflow-hidden ${styles['player-wrapper']}`}
+        >
+          <ReactPlayer
+            className={`${styles['react-player']}`}
```
Comment (patheticGeek) [inline src/components/Videos/Video.js:26]: Can be better as `className={styles['react-player']}`

Comment (patheticGeek) [reply src/components/Videos/Video.js:26]: Do this in other places too

File: src/pages/Videos.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+          .catch((err) => {
+            toast.error(err.message);
+            setIsLoading(false);
+          });
+      })
+      .catch((err) => {
+        toast.error(err.message);
+        setIsLoading(false);
+      });
+  }, []);
```
Comment (patheticGeek) [inline src/pages/Videos.js:88]: Inside use effect create a async function and call that 
Always use async/await instead of promises

**Review by patheticGeek — CHANGES_REQUESTED** [review]
The amount of custom CSS used can be reduced significantly, use bootstrap classes instead of custom ones and as there is flex use it's properties rather than writing media queries

File: src/components/Videos/Video.js (hunk 2 above)
Comment (sanjay270899) [reply src/components/Videos/Video.js:26]: If we have to add any other class then we can add it direct without taking care of adding extra curly brackets.

**Pull request merged**